### ISSUE TITLE
[Bug] token-classification return start=null and end=null instead of index in original text

### ISSUE THREAD

**Issue by ldenoue** [xenova/transformers.js#359]
**Describe the bug**
*A clear and concise description of what the bug is.*

**How to reproduce**
*Steps or a minimal working example to reproduce the behavior*

let classifier = await pipeline('token-classification', 'Xenova/bert-base-NER');
let output = await classifier('My name is Sarah and I live in London');

[
    {
        "entity": "B-PER",
        "score": 0.9980159401893616,
        "index": 4,
        "word": "Sarah",
        "start": null,
        "end": null
    },
    {
        "entity": "B-LOC",
        "score": 0.9994593262672424,
        "index": 9,
        "word": "London",
        "start": null,
        "end": null
    }
]

**Expected behavior**

I expected "start" and "end" to point to the start and end in the original text.

**Environment**
- Transformers.js version: https://cdn.jsdelivr.net/npm/@xenova/transformers@2.6.2
- Browser (if applicable): Chrome
- Operating system (if applicable): MacOS
- Other:

**Additional context**

The documentation for the python version gives back proper start and end points.
https://huggingface.co/docs/transformers/main_classes/pipelines#transformers.TokenClassificationPipeline.example

**Comment by xenova** [comment]
Hi there 👋 The reason for this is that we haven't yet added this functionality 😅  (so, less of a bug and more of a feature request). The reason for this was - at the time - I didn't need it for a demo I was creating, and I didn't find time to go back and fix it.

Luckily there is an open [PR](https://github.com/xenova/transformers.js/pull/333) for this (+ first feature request [here](https://github.com/xenova/transformers.js/issues/312))

**Comment by llmer** [comment]
Here's a pr that fixes this issue: https://github.com/xenova/transformers.js/issues/359

**Comment by joprice** [comment]
@xenova I see an attempt was made to implement this one for the token pipeline but was closed https://github.com/xenova/transformers.js/pull/508 due to discrepancies. I tried this change out and it works well for my model. I checked out the transformers implementation, but it was a bit hard to follow at a glance as it was handling overlapping tokens and a few other cases, and I didn't spent enough time on it. Any tips on how to approach and test this? If it's just following the python implementation and tests I can go for dig into that, but perhaps something differs between the raw transformers side and the post-onnx conversion side.
